### DISCUSSION TITLE
Hefty performance impact to be expected?

### DISCUSSION THREAD

**Discussion by Milliw** [royshil/obs-backgroundremoval#424] [General]
Hello guys and girls! ((c) Mutahar ^^)

While this plugin is really very useful to me I seem to can not use it for live streaming due to its strong performance impact to my system/the stream. And I would like to know if this is to be expected or if I am encountering some odd behaviour here.

My system:
Nobara 38 (based on Fedora 38), AMD Radeon RX 5700, AMD Ryzen 9 3900X, 32 GB 3200 MHz RAM.

1. Somehow weird thing is, that when upping the used cpu thread setting more than 2 the cpu usage goes up very high though not delivering any better performance (e.g. frame render time) but instead getting even worse.
2. And although cpu usage of OBS overall is "only" about ~15% when filter enabled, it leads to large loss in gaming performance going from locked 100 fps down to sometimes ~70 fps. And that is while not streaming/recording at all. Just enabling background blur. GPU usage goes then down which indicates cpu bottleneck.

What do you say?
Are you able to use that plugin without any/such issues?

EDIT:
I just watched the latest YT video of you explaining the plugin and I noticed that there is no such "restless" border that I have. Or at least it is not noticeable?! (Example what I mean can be seen here: https://www.twitch.tv/videos/1899560048?t=00h34m58s)

**Comment by royshil** [comment]
hi @Milliw 
thanks for the good feedback
what BG model are you using? 
"Mediapipe" should be the fastest one, it often works for me on CPU (got Mac Intel) with 1 thread and takes just ~5-8% CPU only
you can also try the "TC Mono Depth" model which is very fast, but it's not a BG segmentation model - it's a depth model so you should tweak the threshold settings

since your system is AMD based (e.g. not Nvidia or Intel) the hardware acceleration of the neural network is limited...
i've looked into ROCm and OpenCL acceleration but haven't been able to strike a balance yet.
I will look into it again to see if there's potential to improve.

re the "restless border" effect - i think since these models are supposed to be used for video conferencing like Zoom and Google Meet - your angle and the chair are not something it expects and so it flickers back and forth. 
but if there's flicker - test the "threshold" setting. set it all the way down to e.g. 0.05
try that out
